js/news.js

```diff
@@ -0,0 +1,59 @@
+"use strict";
+class fetchdata {
+  apiFetcher(url) {
```

Comment:
`fetchdata` class was expected to be a generic base class that is responsible for fetching any data by `url` and return json. here you have a very specific `apiFetcher` method (a strange name for a method) that has specific logic related with `newsSource` instance